catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -70,6 +70,22 @@
 
     private static final String DISTANCE_SORT_FIELD = "_distance_";
 
+    private static final String SOLR_QUERY_ROWS_PROPERTY = "solr.query.rows";
+
+    /**
+     * Because {@link SolrQuery#setRows(Integer)} will increment the given integer and use it as an
+     * array index in Solr Cloud mode, we cannot use {@link Integer#MAX_VALUE} by default and should
+     * pick a lower, sensible ceiling for our row count. In practice, correct paging techniques should
+     * be implemented; if this is not the case, the default row count of 2 billion can be overridden
+     * in system properties to prevent a browser crash or any fatal client-side error at the expense
+     * of missing query results.
+     *
+     * (See DDF-2872)
+     */
+    private static final Integer SOLR_QUERY_DEFAULT_ROWS = 2000000000;
```

Comment:
It should have been before the hero state 6 days ago. A non functional change is fine to make even in the middle of heroing since we know without a doubt this will not affect the code behavior in any way.

I can't figure out what that number is without counting zeros, so it would be a big help to have something to help understand what this number is.